### PR TITLE
Impleted testings 

### PR DESCRIPTION
**Pull request by thedekerone**
In this PR we implemented testing methods for the module Enumerable.

- We used Rspec to build the tests.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,219 @@
+require './my_each.rb'
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: [Optional]
when you are requiring a file it is best practice to use, `require_relative` so the line should look like,
```ruby
require_relative '../my_each.rb'
```

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    context 'when applied block of code' do
+      it 'must return the same array' do
+        expect(array.my_each { |x| x + 2 }).to eql(array)
+      end
+
+      it 'must execute for every element' do
+        test_array = []
+        array.my_each { |x| test_array.push(x + 2) }
+        expect(test_array).to eql([3, 4, 5])
+      end
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: [Optional]
must execute what? I would say, 'calls the given block for each element'
And you can make this test more solid, by comparing it with the actual `each` method. I just modified your current test into this format,
```ruby
it 'calls the block for each element' do
  test_array = []
  block = proc { |x| test_array.push(x + 2) }
  array.each(&block)
  actual_output = test_array.dup
  test_array = []
  array.my_each(&block)
  expect(test_array).to eql(actual_output)
end
```

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,219 @@
+require './my_each.rb'
+
+describe Enumerable do
+  let(:array) { Array.new([1, 2, 3]) }
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: [Optional]
`Array.new` actually no needed. just [1, 2, 3] will do it. Now seems to me like you are trying to create a copy of an array. That is not needed.

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+    context 'When the block is not given and parameter is given:' do
+      it 'regex' do
+        expect(%w[ant bear cat].my_all?(/t/)).to eql(false)
+      end
+      it 'Class' do
+        expect([1, 2i, 3.14].my_all?(Numeric)).to eql(true)
+      end
+    end
+    context 'When the neither the block or parameter is given' do
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: did you mean, When neither the block or parameter is given?

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        expect([].my_all?).to eql(true)
+      end
+      it 'the array is not empty' do
+        expect([nil, true, 99].my_all?).to eql(false)
+      end
+    end
+  end
+  describe '#my_any?' do
+    context 'When the block is given' do
+      it 'any element matches the condition in block' do
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: [Optional]
You can organize the message like this way
`returns true if the block returns true for any element`

File: spec/enumerable_spec.rb [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        expect(%w[ant bear cat].my_any? { |word| word.length > 3 }).to eql(true)
+      end
+
+      it 'none of the elements matches the condition in block' do
```
Comment (shshamim63) [inline spec/enumerable_spec.rb]: [Optional]
same as the above :point_up: .

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Changes Required :tractor: 
___
Hi @thedekerone and @shubhsk88 
Awesome job. Tests look well- organized, you have used let to define variables. Also added test for methods which should raise an error. Amazing :clap: . But just found some simple corner cases, which are holding me back to approve it now. And it is really easy to cover them. Let's get to them.
___
### :spades: Mandatory Changes :spades: 
___
1. I can not see any info related to the test in your Readme. Can you add some instruction like which testing tools you have used (obviously Rspec). And how to execute them.
2. I think some negative scenarios are needed to the following methods
- my_all?
- my_any?
- my_none?
the negative cases like when it should return false if you have a case where it returns true. Add them for the following cases
- when class is given
- when a regex is given
___
### :clubs: Optional Changes :clubs: 
___
- You can update some of your tests to compare them with the actual methods. I have added a sample. Well, that will help you to get an overview.
- I also added some suggestions with your test block messages. I will suggest you update some of your messages so that it can deliver the actual test criteria.
___
##### Please fix the issues mentioned above and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by mosesogwo — APPROVED** [review]
**Congratulations, your project is approved:+1:**

**Good luck in your next project :stars:**

### Status: Approved:heavy_check_mark:
